datafusion/proto/src/logical_plan.rs

```diff
@@ -491,14 +461,23 @@ impl AsLogicalPlan for LogicalPlanNode {
                     ))
                 })?;
 
-                let pb_file_type: protobuf::FileType =
-                    create_extern_table.file_type.try_into()?;
+                match create_extern_table.file_type.as_str() {
+                    "CSV" | "JSON" | "PARQUET" | "AVRO" => {}
```

Comment:
It might be nice to avoid having these four strings hardcoded in several places, so that if we add another format, we don't have to make changes all over the place.

Maybe we could put it into a struct like `BuiltInTableProviders` or something

Reply:
I 100% agree, but I was hoping to do it in a follow-on PR, as there seems to be a lot of action going on with TableProviders right now, and I was hoping to not keep this one long-standing.